### ISSUE TITLE
Pretrained GoogLeNet without batch norm

### ISSUE THREAD

**Issue by addisonklinke** [pytorch/vision#2634]
## 🚀 Feature
Looking at the [commit history](https://github.com/pytorch/vision/commits/e1a3042792a3f15a8cae8cc563d54d2215a3e201/torchvision/models/googlenet.py) for `torchvision.models.googlenet`, I noticed that the squashed commit messages for the initial PR [#678](https://github.com/pytorch/vision/pull/678/commits/6c0d34d250bbc85a556f58f6aed8190ed14e79eb) mention a plain _and_ batch norm version of GoogLeNet. Currently, there is only one `googlenet` defined in that module's `__all__`. Is there a reason we can't offer both `googlenet` and `googlenet_bn` like before?

## Motivation

I am trying to use PyTorch to replicate a custom Caffe model which used GoogLeNet as one of its components. In both the Caffe and ONNX implementation GoogLeNet, batch norm is not used. In order to have the most direct comparison, I would like to have access to a PyTorch model which also does not use batch norm. 

## Pitch

Use the commit history of the linked PR to supply two GoogLeNet implementations - one with batch norm and one without

## Alternatives

As mentioned in the [forums](https://discuss.pytorch.org/t/how-to-delete-layer-in-pretrained-model/17648/2), I could use `nn.modules.Linear.Identity` to delete all batch norm layers from the current implementation. However, this would render the pre-trained weights ineffective and therefore require many more epochs to get an optimized backbone. With a smaller dataset, I am also not confident that the quality of the backbone would ever reach that of the pre-trained ImageNet model.

## Additional context

The official ONNX GoogLeNet can be downloaded from [here](https://github.com/onnx/models/tree/master/vision/classification/inception_and_googlenet/googlenet/model). Since I am using, PyTorch 1.4.0, I chose the ONNX opset 6. Following your conversion [docs](https://pytorch.org/tutorials/advanced/super_resolution_with_onnxruntime.html), I wrote the following script to save PyTorch's GoogLeNet as an ONNX file

```
from numpy.testing import assert_allclose
import onnx
import onnxruntime
import torch
from torchvision.models import googlenet
import torch.onnx


def to_numpy(tensor):
    return tensor.detach().cpu().numpy() if tensor.requires_grad else tensor.cpu().numpy()


# Trace inputs through the PyTorch model to export
model = googlenet(pretrained=True)
model.eval()
dummy = (torch.randn(1, 3, 224, 224), )
torch_outs = model(*dummy)
path = 'googlenet-pytorch.onnx'
torch.onnx.export(
    model=model,
    args=dummy,
    f=path,
    do_constant_folding=True,
    input_names=['images'],
    output_names=['scores'],
    dynamic_axes={
        'images': {0: 'batch_size'},
        'scores': {0: 'batch_size'}})

# Load/check the ONNX model and run through Python API
onnx_model = onnx.load(path)
onnx.checker.check_model(onnx_model)
ort_session = onnxruntime.InferenceSession(path)
ort_inputs = {inp.name: to_numpy(dummy[i]) for i, inp in enumerate(ort_session.get_inputs())}
ort_outs = ort_session.run(None, ort_inputs)

# Confirm outputs match
assert_allclose(to_numpy(torch_outs), ort_outs[0], rtol=1e-03, atol=1e-05)
```

Loading both implementations into [Netron](https://github.com/lutzroeder/netron) for visualization, you can clearly see how the first inception block differs in regard to its use of batch norm

**PyTorch**

![image](https://user-images.githubusercontent.com/39926682/91772910-fec87200-eba2-11ea-8bff-450e0bb07af9.png)

**ONNX**

![image](https://user-images.githubusercontent.com/39926682/91772978-1ef83100-eba3-11ea-9640-5a9b549a3790.png)

**Comment by vfdev-5** [comment]
@addisonklinke thanks for FR. 

@TheCodez could you please shed some light why `googlenet_bn` was not retained ?

**Comment by TheCodez** [comment]
The original model I send was completely based off the paper and used the same initialization that inception_v3 uses, which is the same the tensorflow implementation uses. 
The tensorflow implementation uses batch norm. I'm guessing the original googlenet doesnt work with this initialization without batchnorm. 

After those failed training attemps I added the Xavier initialization as that worked for the BVLC Googlenet. Sadly this version was never trained by @fmassa, as we decided to use the pretrained weights from tensorflow instead, which network structure is a bit different from what the paper describes https://github.com/pytorch/vision/issues/906

**Comment by addisonklinke** [comment]
@TheCodez So if I understand correctly, your initial work _did_ fully implement the architecture for a non-batch norm GoogeLeNet, but we never had pre-trained weights for it? Since the weights are from Tensorflow, we had to use their batch norm based architecture

**Comment by TheCodez** [comment]
@addisonklinke Correct. Given that training Googlenet from scratch is known for being hard, using the Tensorflow weights and architecture was the easier choice.

**Comment by addisonklinke** [comment]
@vfdev-5 Since training is the issue, perhaps we could port pre-trained weights for the non-batch norm architecture from the official ONNX model I linked?

**Comment by vfdev-5** [comment]
@TheCodez could you please add a link on the tensorflow implementation ? 

@addisonklinke there are also the weights from BAIR caffe : https://github.com/BVLC/caffe/tree/master/models/bvlc_googlenet which can be also ported. This one is without BN and has conv kernel=5 for branch3 (#906).

I also wonder how would we call this "original" googlenet here keeping current implementation with its name for backward compatibility.

**Comment by TheCodez** [comment]
@vfdev-5 https://github.com/tensorflow/models/blob/master/research/slim/nets/inception_v1.py

That's a good question, ideally the current googlenet should be called inception_v1 but this makes backwards compatibility hard.

**Comment by vfdev-5** [comment]
I think it worth to update the docs of https://github.com/pytorch/vision/blob/195c41ee7ae3fff8f64a4fda67632259dd4f7454/torchvision/models/googlenet.py#L26 saying that implementation and weights are from tf-slim.

Maybe, "original" model can be called as `bvlc_googlenet` and implementation's doc should also contain a link on caffe repository and both`bvlc_googlenet` and `googlenet` should note the difference between each other.

@addisonklinke would you be interested in implementing `bvlc_googlenet` and porting caffe weights to pytorch ?

cc @fmassa for visibility and other thoughts

**Comment by addisonklinke** [comment]
@vfdev-5 I'm on a rather time-sensitive project, so will likely pursue some alternatives like using a different `torchvision.models` class for the backbone rather than diving down the Caffe to PyTorch rabbit hole. If that doesn't pan out, then I may have more time and effort to invest into porting from another framework
